src/OrchardCore/OrchardCore.Data.YesSql/SqliteHelper.cs

```diff
@@ -24,7 +34,28 @@ public static string GetConnectionString(SqliteOptions sqliteOptions, string dat
             : Path.Combine(databaseFolder, databaseName),
             Cache = SqliteCacheMode.Shared,
             Pooling = sqliteOptions.UseConnectionPooling,
-            Mode = createFile ? SqliteOpenMode.ReadWriteCreate : SqliteOpenMode.ReadWrite
+            Mode = shellSettings.IsInitializing() ? SqliteOpenMode.ReadWriteCreate : SqliteOpenMode.ReadWrite
+        }
+        .ToString();
+    }
+
+    /// <summary>
+    /// Generic helper for any other SQLite connection. Restricted to ReadWrite.
+    /// </summary>
+    /// <param name="sqliteOptions"></param>
+    /// <param name="databaseFolder"></param>
+    /// <param name="databaseName"></param>
+    /// <returns></returns>
+    public static string GetConnectionString(SqliteOptions sqliteOptions, string databaseFolder, string databaseName)
```

Comment:
The body of this method can be refactored not to copy from the other one, but have a single private method with a `SqliteOpenMode` parameter.

Reply:
Not sure what you mean by private but I added the param to this method as suggested.

Reply:
This code:

```c#
        return new SqliteConnectionStringBuilder
        {
            DataSource = string.IsNullOrEmpty(databaseFolder)
            ? databaseName
            : Path.Combine(databaseFolder, databaseName),
            Cache = SqliteCacheMode.Shared,
            Pooling = sqliteOptions.UseConnectionPooling,
            Mode = <this is different>
        }
        .ToString();
```

Is repeated in this class. What I'm suggesting is having a private method used by both public ones that does this, with a parameter for `Mode`. However, this is good too, but then the first method should use the second one.

Reply:
Ok I'll take a look.